src/refmt/eol_detect.ml

```diff
@@ -0,0 +1,19 @@
+type eol =
+  | LF 
+  | CRLF 
+let show eol = match eol with | LF -> "lf" | CRLF -> "crlf"
+let default_eol = match Sys.win32 with | true -> CRLF | _ -> LF
+let crlf_regex = Str.regexp ".*\r\n$"
+let get_eol_from_string s =
+  try ignore (Str.search_forward crlf_regex s 0); CRLF with | Not_found -> LF
```

Comment:
Btw, in this version, instead of `CRLF` shouldn't it be the default char for the platform?

Reply:
The way I designed the solution here was that this `eol_detect` module reports the current state of the world (ie, whether the string/file has a `CRLF` or not) - no translation is happening at this point. Perhaps it could be named better to reflect th is?

IMO, it'd be confusing if `get_eol_from_string "test\r\n"` returned `LF` on Linux/OSX. 

We then decide how to act on that information elsewhere (like, whether we should use `open_out` or `open_out_bin`). Let me know if you have ideas to improve this!